frontend/src/components/ui/tooltip.tsx

```diff
@@ -25,4 +43,58 @@ const TooltipContent = React.forwardRef<
 ));
 TooltipContent.displayName = TooltipPrimitive.Content.displayName;
 
-export { Tooltip, TooltipTrigger, TooltipContent, TooltipProvider };
+// Hybrid tooltip
+
+const TouchContext = createContext<boolean | undefined>(undefined);
+const useTouch = () => useContext(TouchContext);
+
+const TouchProvider = (props: PropsWithChildren) => {
+  const [isTouch, setTouch] = useState<boolean>();
+
+  useEffect(() => {
+    setTouch(window.matchMedia("(pointer: coarse)").matches);
+  }, []);
+
+  return <TouchContext.Provider value={isTouch} {...props} />;
+};
+
+const HybridTooltip = (props: TooltipProps & PopoverProps) => {
+  const isTouch = useTouch();
+
+  return isTouch ? <Popover {...props} /> : <Tooltip {...props} />;
+};
+
+const HybridTooltipTrigger = (
+  props: TooltipTriggerProps & PopoverTriggerProps
+) => {
+  const isTouch = useTouch();
+
+  return isTouch ? (
+    <PopoverTrigger {...props} />
+  ) : (
+    <TooltipTrigger {...props} />
+  );
+};
+
+const HybridTooltipContent = (
+  props: TooltipContentProps & PopoverContentProps
+) => {
+  const isTouch = useTouch();
+
+  return isTouch ? (
+    <PopoverContent
+      {...props}
+      className={cn("bg-primary text-primary-foreground", props.className)}
```

Comment:
I think we should reduce the font-size a bit, in desktop we do `text-sm` for title (Spending Balance) and `text-xs` for the tooltip content, for mobile I think `text-sm` should be ideal

<table>
<tr>
<td>
Current
</td>
<td>
text-sm
</td>
<td>
text-xs
</td>
</tr>
<tr>
<td>
<img src="https://github.com/user-attachments/assets/6e83f228-92fc-4045-a147-920f21bea3d9"/>
</td>
<td>
<img src="https://github.com/user-attachments/assets/4bca5d40-4a4d-444b-8776-463ec1a80aa6"/>
</td>
<td>
<img src="https://github.com/user-attachments/assets/a1221a0d-2b14-49c9-bcf6-a40016206bd1"/>
</td>
</tr>
</table>